src/KM_Settings.pas

```diff
@@ -970,4 +944,122 @@ function TKMWindowParams.IsValid(aMonitorsInfo: TKMPointArray): Boolean;
 end;
 
 
+{TKMFavouriteMaps}
+constructor TKMFavouriteMaps.Create;
+begin
+  inherited;
+  fFavouriteMPMaps := TStringList.Create;
+  fFavouriteMPMaps.Delimiter       := FAVOURITE_MAPS_DELIMITER;
+  fFavouriteMPMaps.StrictDelimiter := True; // Requires D2006 or newer.
+end;
+
+
+destructor TKMFavouriteMaps.Destroy;
+begin
+  FreeAndNil(fFavouriteMPMaps);
+  inherited;
+end;
+
+
+procedure TKMFavouriteMaps.FavoriteMapsUpdated;
+begin
+  if Assigned(fOnMapsUpdate) then
+    fOnMapsUpdate(PackToString);
+end;
+
+
+procedure TKMFavouriteMaps.LoadFromString(aString: UnicodeString);
+var I: Integer;
+    MapCRC : Int64;
+    StringList: TStringList;
+begin
+  fFavouriteMPMaps.Clear;
+  StringList := TStringList.Create;
+  StringList.Delimiter := FAVOURITE_MAPS_DELIMITER;
+  StringList.DelimitedText   := Trim(aString);
+
+  for I := 0 to StringList.Count - 1 do
+  begin
+    if TryStrToInt64(Trim(StringList[I]), MapCRC)
+      and (MapCRC > 0)
+      and not Contains(Cardinal(MapCRC)) then
+      fFavouriteMPMaps.Add(Trim(StringList[I]));
+  end;
+
+  StringList.Free;
+end;
+
+
+function TKMFavouriteMaps.PackToString: UnicodeString;
+begin
+  Result := fFavouriteMPMaps.DelimitedText;
+end;
+
+
+//Sync current Favourites Maps list with maps CRC array.
+procedure TKMFavouriteMaps.SyncWithMaps(aMapsCRCArray: TKMCardinalArray);

```

Comment:
I find the name Sync confusing, it implies some kind of mirroring. Maybe RemoveNonExistentFavourites or something like that would be more descriptive?